### PR TITLE
AST -> PolyType for Record

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: libflux/src/core/semantic/convert.rs [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 use crate::semantic::fresh::Fresher;
 use crate::semantic::nodes::*;
+use crate::semantic::types;
 use crate::semantic::types::MonoType;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
 use std::result;
 
 pub type SemanticError = String;
+pub type SemanticMap<K, V> = BTreeMap<K, V>;
```
Comment (jlapacik) [inline libflux/src/core/semantic/convert.rs:11]: You should just be able to import this from the other file.

File: libflux/src/core/semantic/convert.rs [hunk 2]
```diff
@@ -138,6 +142,97 @@ fn convert_builtin_statement(stmt: ast::BuiltinStmt, fresher: &mut Fresher) -> R
     })
 }
 
+pub type MonoTypeMap = SemanticMap<String, MonoType>;
```
Comment (jlapacik) [inline libflux/src/core/semantic/convert.rs:145]: Similarly can you re-use the MonoTypeMap from `types.rs`?

File: libflux/src/core/semantic/convert.rs [hunk 3]
```diff
@@ -138,6 +142,97 @@ fn convert_builtin_statement(stmt: ast::BuiltinStmt, fresher: &mut Fresher) -> R
...
+            "bool" => Ok(MonoType::Bool),
+            "int" => Ok(MonoType::Int),
+            "uint" => Ok(MonoType::Uint),
+            "float" => Ok(MonoType::Float),
+            "string" => Ok(MonoType::String),
+            "duration" => Ok(MonoType::Duration),
+            "time" => Ok(MonoType::Time),
+            "regexp" => Ok(MonoType::Regexp),
+            "bytes" => Ok(MonoType::Bytes),
+            _ => Err("Bad parameter type.".to_string()),
```
Comment (jlapacik) [inline libflux/src/core/semantic/convert.rs:167]: Can you use a better error message here? Like `"invalid named type {}", ...`

File: libflux/src/core/semantic/convert.rs [hunk 4]
```diff
@@ -138,6 +142,97 @@ fn convert_builtin_statement(stmt: ast::BuiltinStmt, fresher: &mut Fresher) -> R
...
+                            _pipe = Some(types::Property {
+                                k: match name {
+                                    Some(n) => n.name,
+                                    None => String::from("<-"),
+                                },
+                                v: convert_monotype(monotype, tvars, f)?,
+                            });
+                            dirty = true;
+                        } else {
+                            return Err("Bad parameter type.".to_string());
```
Comment (jlapacik) [inline libflux/src/core/semantic/convert.rs:198]: Can you also use a better error message here? Like `"function types can have at most one pipe parameter"`

File: libflux/src/core/semantic/convert.rs (hunk 2 above)
Comment (ChitlangeSahas) [reply libflux/src/core/semantic/convert.rs:145]: ✅

File: libflux/src/core/semantic/convert.rs (hunk 1 above)
Comment (ChitlangeSahas) [reply libflux/src/core/semantic/convert.rs:11]: ✅

**Pull request merged**